### PR TITLE
Ignore DisallowedHosts logs from Sentry

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description

Ignores the logging mentioned in the comment as it's completely un-actionable.

## Testing Instructions

Run the API locally and verify that you can make requests without issue.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
To whichever Openverse maintainer with Sentry access, please document the resolution when this is committed to main on the Sentry issue so that we don't keep getting alerts for this: https://sentry.io/share/issue/9af3cdf8ef74420aa7bbb6697760a82c/

File: api/catalog/settings.py [hunk 1]
```diff
@@ -350,3 +351,9 @@
         send_default_pii=False,
         environment=ENVIRONMENT,
     )
+
+    # ALLOW_HOSTS is correctly configured so ignore this to prevent
+    # spammy bots like https://github.com/robertdavidgraham/masscan
+    # from pushing un-actionable alerts to Sentry like
+    # https://sentry.io/organizations/openverse/issues/3252810791
```
Comment (krysal) [inline api/catalog/settings.py:358]: Should this be the public URL instead? Openverse maintainers can go to the details from there.

**Review by krysal — COMMENTED** [review]
Can we disallow that host in particular? `openverse-api-alb-605315325.us-east-1.elb.amazonaws.com`
I haven't seen another spammy address but if we silence the whole exception, it could be hard to debug if there are infrastructure changes in the future.

**Comment by sarayourfriend** [comment]
> Can we disallow that host in particular? openverse-api-alb-605315325.us-east-1.elb.amazonaws.com

All hosts other than the allowed hosts are already disallowed. And the ALB DNS name is liable to change whenever the ALB is redeployed anyway.

This is only excluding this from Sentry (it is truly unactionable from a Sentry perspective) so we will still see this error in the logs. If we're trying to debug infrastructure changes we'll still be able to see this information in the logs, it just won't keep getting sent to Sentry where it's effectively useless information outside of the edge case of when we're configuring a new deployment and therefore actively watching the logs and debugging this kind of thing.

**Review by krysal — APPROVED** [review]
You're right! Forgot this is only to silence it from Sentry, we still have Django logs, so this is great. Thanks!

**Review by zackkrida — APPROVED** [review]
Great idea

**Pull request merged**